### PR TITLE
[multimodal] Bound tokenizer-side GPU memory in Kimi GPU image preprocessing

### PR DESCRIPTION
**Pull request by chromecast56**
## Motivation

`_gpu_preprocess_images` (Kimi K2.5/K3) keeps every image's fp32 patch tensor alive in a list and ends with a request-wide `torch.cat`. Peak GPU memory therefore grows with the request's total patch data because the accumulated outputs coexist with the final concatenation. On deployments running a high `--mem-fraction-static`, a large multi-image request can OOM device 0. The CUDA caching allocator may retain the resulting high-water reservation for reuse, reducing headroom for later requests.

## Modifications

- `_gpu_preprocess_images`: process size-groups in sub-batches capped by `SGLANG_MM_GPU_PREPROCESS_CHUNK_BYTES` (default 512 MB); return per-image tensors instead of a request-wide concat; optionally hand each image to a `per_image_sink` as it is produced.
- `MMFeatureStreamSink` (K2.5/K3): hash each image and wrap it into the CUDA-IPC pool (or move it to host for non-IPC transports) at production time, so preprocessing no longer retains every image's GPU output at once.
- `get_new_expanded_mm_items`: split list-shaped bundled features by element and preserve production-time hashes (`PRECOMPUTED_FEATURE_HASHES_KEY`) instead of re-hashing wrapped features.
- Kimi-K3's deferred model-side GPU preprocessing path materializes the per-image list before its grouped feature validation and splitting.

`mimo_v2` has the same accumulate-and-concat pattern and can be ported in a follow-up.

## Validation

Outputs are bit-exact with the previous pipeline and hashes are unchanged (same bytes and algorithm), so radix-cache keys remain stable. `test/registered/unit/multimodal/test_kimi_mm_preprocess_bounded.py` checks parity across size groups, chunking, and K3 RGBA compositing.

On H100, `test_peak_memory_bounded` measures an unchunked peak of at least 2× the total patch bytes and a bounded-path peak of at most 4× the configured chunk size for the test workload. Kernels are unchanged; chunking only splits batched launches.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations) (no user-facing interface beyond the environment variable).
- [x] Provide accuracy and memory validation results.
- [ ] Provide a speed benchmark (kernel math is unchanged; out of scope for this memory-bound change).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

---
### CI States

Latest PR Test (Base): :x: [Run #34399129594](https://github.com/sgl-project/sglang/actions/runs/34399129594)
Latest PR Test (Extra): :x: [Run #34399129196](https://github.com/sgl-project/sglang/actions/runs/34399129196)
Latest PR Test (AMD ROCm 10): :x: [Run #34399129583](https://github.com/sgl-project/sglang/actions/runs/34399129583)

### CODE REVIEW

**Comment by gongy** [comment]
/rerun-test test_kimi_mm_preprocess_bounded.py test_mm_utils_split.py test_kimi_k3_gpu_preprocess.py test_cuda_ipc_transport.py test_cuda_ipc_pool_budget.py test_feature_materialization.py test_base_processor_image_decode.py test_vision_openai_server_a.py test_encoder_dp.py test_vlm_input_format.py test_vision_chunked_prefill.py

**Comment by rodamani** [comment]
Rebased onto main and resolved conflicts in managers/mm_utils.py, processors/base_processor.py, test_kimi_k3_vision.py. Also fixed two issues from the per-image return type.
1. processors/kimi_k3.py::prepare_image_features still called .shape[0] on the list
2. Two _gpu_preprocess_images mocks in test_kimi_k25.py returned a bare tensor

Test_mm_utils_split, test_kimi_k25, test_kimi_k3_vision pass locally. test_kimi_mm_preprocess_bounded needs GPU, so checking with CI. Ran pre-commit on changed files. cc @chromecast56
